imapclient/notify_test.go

```diff
@@ -0,0 +1,357 @@
+package imapclient_test
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/emersion/go-imap/v2"
+	"github.com/emersion/go-imap/v2/imapclient"
+)
+
+func TestClient_Notify(t *testing.T) {
+	if os.Getenv("GOIMAP_TEST_DOVECOT") != "1" {
+		t.Skip("NOTIFY test requires GOIMAP_TEST_DOVECOT=1")
+	}
+
+	existsCh := make(chan uint32, 1)
+
+	// Clients needs a UnilateralDataHandler configured, so set it up manually.
+	conn, server := newDovecotClientServerPair(t)
+	defer server.Close()
+
+	options := &imapclient.Options{
```

Comment:
Oh, do we need to pass custom `UnilateralDataHandler` to the client, but `newClientServerPair` doesn't support this?